File: Algorithms/Recursion/tower_of_hanoi.c
#include<stdio.h>


void tower_of_hanoi(int n, char from, char to, char temp) {
	if(n == 1) {
		printf("Move Disc 1: %c -> %c\n", from, to);
		return;
	}

	tower_of_hanoi(n-1, from, temp, to);
	printf("Move Disc %d: %c -> %c\n", n, from , to);
	tower_of_hanoi(n-1, temp, to , from);
}

int main() {
	int n;
	scanf("%d", &n);
	tower_of_hanoi(n, 'A', 'C', 'B');
	return 0;
}

File: Algorithms/Recursion/tower_of_hanoi.py


def tower_of_hanoi(n, fromm, to, temp):
    if n==1:
        print(f'move disk to 1: {fromm} -> {to}')
        return
    
    tower_of_hanoi(n-1, fromm, temp, to)
    print(f'move disk to {n} : {fromm} -> {to}')
    tower_of_hanoi(n-1, temp, to, fromm)


if __name__ == "__main__":
    n = int(input("Enter a number"))
    tower_of_hanoi(n, "A", "B", "C")
